addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderServerBridgeHandler.java

```diff
@@ -123,211 +144,173 @@ public void run() {
-                    getLogIdentifier(), refreshCycleCount, refreshFrequency, iMaxCycles,
-                    getBridgeConfig().getRefreshInterval(), getBridgeConfig().getRefreshIntervalLowPriorityTask());
+                refreshThing(cyclePriority);
 
-            if (doRefresh) {
-                if (resetCount == true) {
+                if ((refreshCycleCount >= intervalLow) && (refreshCycleCount >= intervalNormal)
+                        && (refreshCycleCount >= intervalHigh) && (refreshCycleCount >= intervalBatch)) {
                     refreshCycleCount = 0;
                 }
 
-                logger.debug("{}: 'refreshDataRunnable()': (diskUsage='{}')", getLogIdentifier(), fetchDiskUsage);
-
-                refreshThing(zoneMinderSession, fetchDiskUsage);
-            }
-        } catch (Exception exception) {
-            logger.error("{}: monitorRunnable::run(): Exception: ", getLogIdentifier(), exception);
-        }
-    };
-
-    private Runnable refreshPriorityDataRunnable = () -> {
-        try {
-            // Make sure priority updates is done
-            for (Thing thing : getThing().getThings()) {
-                try {
-                    if (thing.getThingTypeUID().equals(ZoneMinderConstants.THING_TYPE_THING_ZONEMINDER_MONITOR)) {
-                        Thing thingMonitor = thing;
-
-                        ZoneMinderBaseThingHandler thingHandler = (ZoneMinderBaseThingHandler) thing.getHandler();
-                        if (thingHandler != null) {
-
-                            if (thingHandler.getRefreshPriority() == DataRefreshPriorityEnum.HIGH_PRIORITY) {
-                                logger.debug("[MONITOR-{}]: RefreshPriority is High Priority",
-                                        thingHandler.getZoneMinderId());
-                                thingHandler.refreshThing(zoneMinderSession, DataRefreshPriorityEnum.HIGH_PRIORITY);
-                            }
-                        } else {
-                            logger.debug(
-                                    "[MONITOR]: refreshThing not called for monitor, since thingHandler is 'null'");
-                        }
-                    }
-                } catch (NullPointerException ex) {
-                    // This isn't critical (unless it comes over and over). There seems to be a bug so that a
-                    // null
-                    // pointer exception is coming every now and then.
-                    // HAve to find the reason for that. Until thenm, don't Spamm
-                    logger.error("[MONITOR]: Method 'refreshThing()' for Bridge failed for thing='{}' - Exception: ",
-                            thing.getUID(), ex);
-                } catch (Exception ex) {
-                    logger.error("[MONITOR]: Method 'refreshThing()' for Bridge failed for thing='{}' - Exception: ",
-                            thing.getUID(), ex);
-                }
+            } catch (Exception exception) {
```

Comment:
Please catch specific exceptions instead of catching all exceptions. That way it is more clear what exceptions need to be handled and how this is done.